### PR TITLE
VZ 5251: OSD waits for OS to be ready

### PR DESCRIPTION
**Pull request by anders-swanson**
- Uptake new VMO image that waits for OS to be ready before updating OSD
- introduce node-level storage options in the VMO

### CODE REVIEW

File: platform-operator/helm_config/charts/verrazzano/crds/verrazzano.io_verrazzanomonitoringinstances_crd.yaml [hunk 1]
```diff
@@ -16,414 +16,452 @@ spec:
     listKind: VerrazzanoMonitoringInstanceList
     plural: verrazzanomonitoringinstances
     shortNames:
-      - vmi
+    - vmi
     singular: verrazzanomonitoringinstance
   scope: Namespaced
   versions:
-    - name: v1
```
Comment (jmaron99) [inline platform-operator/helm_config/charts/verrazzano/crds/verrazzano.io_verrazzanomonitoringinstances_crd.yaml:23]: These changes are difficult to make sense of here (both in unified and split view).  What is the nature of these changes?

File: platform-operator/helm_config/charts/verrazzano/values.yaml [hunk 2]
```diff
@@ -65,7 +65,7 @@ monitoringOperator:
   defaultSimpleCompReplicas: 1
   defaultPrometheusReplicas: 1
   alertManagerImage: "noimage"
-  esWaitTargetVersion: 7.6.1
+  esWaitTargetVersion: 1.2.3
```
Comment (jmaron99) [inline platform-operator/helm_config/charts/verrazzano/values.yaml:68]: Did we re-introduce the eswait container?

Comment (anders-swanson) [reply platform-operator/helm_config/charts/verrazzano/values.yaml:68]: No, but the VMO takes into account this target version ( i suppose we could rename it ), as a guidance on the Version to look for during upgrade. We are checking this version before updating OSD/Kibana to the new version.

File: platform-operator/helm_config/charts/verrazzano/crds/verrazzano.io_verrazzanomonitoringinstances_crd.yaml (hunk 1 above)
Comment (anders-swanson) [reply platform-operator/helm_config/charts/verrazzano/crds/verrazzano.io_verrazzanomonitoringinstances_crd.yaml:23]: These enable node level storage options to enable PVC resizing without taking down the master (statefulset) nodes.

File: platform-operator/helm_config/charts/verrazzano/values.yaml (hunk 2 above)
Comment (anders-swanson) [reply platform-operator/helm_config/charts/verrazzano/values.yaml:68]: ESWait container will never work with Istio proxy due to technical limitations.

File: platform-operator/helm_config/charts/verrazzano/crds/verrazzano.io_verrazzanomonitoringinstances_crd.yaml (hunk 1 above)
Comment (jmaron99) [reply platform-operator/helm_config/charts/verrazzano/crds/verrazzano.io_verrazzanomonitoringinstances_crd.yaml:23]: They are all modifications/additions to the existing CRD?  No parameters removed?  Just wondering about backwards compatibility...

File: platform-operator/helm_config/charts/verrazzano/values.yaml (hunk 2 above)
Comment (jmaron99) [reply platform-operator/helm_config/charts/verrazzano/values.yaml:68]: OK - that was my understanding (it was no longer usable)

File: platform-operator/helm_config/charts/verrazzano/crds/verrazzano.io_verrazzanomonitoringinstances_crd.yaml (hunk 1 above)
Comment (anders-swanson) [reply platform-operator/helm_config/charts/verrazzano/crds/verrazzano.io_verrazzanomonitoringinstances_crd.yaml:23]: Additions only, I took steps to make it backward compatible. 
See [#157](https://github.com/verrazzano/verrazzano-monitoring-operator/pull/157) and [#156](https://github.com/verrazzano/verrazzano-monitoring-operator/pull/156) in the VMO for additional context.

**Review by jmaron99 — APPROVED** [review]
Approving. I don't see that targeted unit tests/specific acceptance tests are possible for this PR.

**Pull request merged**